### ISSUE TITLE
[iOS] ListView in CarouselView causes System.NullReferenceException

### ISSUE THREAD

**Issue by lightray** [alexrainman/CarouselView#366]
Hello,
I'm developing an application using CarouselViewControl for my mainpage. What I need is to put a ListView for each of the views in the CarouselView. The ContenPage has an AbsoluteLayout, inside of which there is a StackLayout and a ListView. On Android I have no problems doing that, but I can't seem to make it work on iOs, as when I run the app I get this exact same error everytime:

```
System.NullReferenceException: Object reference not set to an instance of an object
 at Xamarin.Forms.Platform.iOS.ListViewRenderer+ListViewDataSource.NumberOfSections (UIKit.UITableView tableView) [0x00000] in <13d193f5a21b4e309afc67e0a5508db5>:0
 at at (wrapper managed-to-native) UIKit.UIApplication:UIApplicationMain (int,string[],intptr,intptr)
 at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Users/builder/data/lanes/5665/db807ec9/source/xamarin-macios/src/UIKit/UIApplication.cs:79
 at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x00038] in /Users/builder/data/lanes/5665/db807ec9/source/xamarin-macios/src/UIKit/UIApplication.cs:63
[...]
```

The funny thing is that I don't really get this ALL the times, more like 9 times out of 10. Sometimes it happened that I was playing around with the code and somehow managed to make it work, only to find out the next time it won't. One time it happened that it worked with the exact same code as my Android version, only to find out it won't, at the next startup. Sometimes it worked removing everything inside the CarouselView but the ListView, but only once - next time it would crash again.

I see NumberOfSections has something to do with ListView grouping, but I'm not using it. If I comment everything out of the ListView and its ViewCell, leaving only the ItemSource property, the error still happens. If I remove the ListView entirely, the code works without any problem. If I wrap the ListView in a CarouselPage instead of a ListView, the code works without any problem.
Any pointers? I'm running out of ideas, been trying this whole day...

I'm running on Visual Studio for Mac, with the latest Xamarin.Forms package (2.5.0.280555) and the latest CarouselView package (5.2.0). I tried deploying against iOs 11.1 and 11.2, on iPhone 8 and iPhone X emulators.

**Comment by AndreiMisiukevich** [comment]
@lightray 
@GamesAgeddon 

Hi) you can try my pack
https://github.com/AndreiMisiukevich/CardView
https://www.nuget.org/packages/CardsView/

**Comment by alexrainman** [comment]
This control has been merged into Xamarin.Forms master branch. Please, wait for it to be released as part of Xamarin.Forms and submit any issues there https://github.com/xamarin/xamarin.forms/tree/feature-carouselview. Thanks

**Comment by LakMoore** [comment]
To fix, nest a StackView between the CarouselView and the ListView.
